aten/src/ATen/native/cuda/Sort.cu

```diff
@@ -46,7 +49,186 @@ bool should_use_th_sort(const Tensor &self, int64_t dim) {
+          (TYPE) keyInfo.strides[collapseKeyDim],                       \
+          valueInfo,                                                    \
+          (TYPE) valueInfo.strides[collapseValueDim],                   \
+          LTComp<scalar_t, true>());                                    \
+      C10_CUDA_KERNEL_LAUNCH_CHECK();                                   \
+    }                                                                   \
+  } while (0)
+
+#define HANDLE_SORT_CASE(TYPE, A)                       \
+  {                                                     \
+    switch (ceilPowerOf2) {                             \
+      case 2048:                                        \
+      HANDLE_CASE(TYPE, A, 2048);                       \
+      break;                                            \
+      case 1024:                                        \
+      case 512:                                         \
+      case 256:                                         \
+      HANDLE_CASE(TYPE, A, 1024);                       \
+      break;                                            \
+      case 128:                                         \
+      case 64:                                          \
+      HANDLE_CASE(TYPE, A, 128);                        \
+      break;                                            \
+      case 32:                                          \
+      case 16:                                          \
+      case 8:                                           \
+      case 4:                                           \
+      case 2:                                           \
+      HANDLE_CASE(TYPE, A, 32);                         \
+      break;                                            \
+      case 1:                                           \
+      /* Nothing to do, data already sorted */          \
+      break;                                            \
+      default:                                          \
+      TORCH_INTERNAL_ASSERT(false);                                    \
+    }                                                   \
+  }
+
+  // The constructed key/value tensor info is used to select the slice
+  // we are sorting on a per-block basis
+  // The constructed key/value tensor info is used to select the slice
+  // we are sorting on a per-block basis
+  AT_DISPATCH_ALL_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, key.scalar_type(), "sortKeyValueInplace", [&]  {
+    if (at::cuda::detail::canUse32BitIndexMath(key)) {
+      at::cuda::detail::TensorInfo<scalar_t, unsigned int> keyInfo =
+        at::cuda::detail::getTensorInfo<scalar_t, unsigned int>(key);
+      at::cuda::detail::TensorInfo<int64_t, unsigned int> valueInfo =
+        at::cuda::detail::getTensorInfo<int64_t, unsigned int>(value);
+
+      if (!key.dim()) {
+        keyInfo.dims = 1;
+        valueInfo.dims = 1;
+        keyInfo.strides[0] = 1;
+        valueInfo.strides[0] = 1;
+      }
```

Comment:
See https://github.com/pytorch/pytorch/pull/55392/files#r619534695 and https://github.com/pytorch/pytorch/pull/55392/files#r619541581 No need to check key.dim() == 0